tests/components/velux/test_config_flow.py

```diff
@@ -0,0 +1,119 @@
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+
+DUMMY_DATA: dict[str, Any] = {
+    CONF_HOST: "127.0.0.1",
+    CONF_PASSWORD: "NotAStrongPassword",
+}
+
+PYVLX_CONNECT_FUNCTION_PATH = "pyvlx.PyVLX.connect"
+PYVLX_DISCONNECT_FUNCTION_PATH = "pyvlx.PyVLX.disconnect"
+
+
+pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_show_form(hass: HomeAssistant) -> None:
+    """Test that the setup form is served."""
+    result: dict[str, Any] = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+
+async def test_user_step_schema(hass: HomeAssistant) -> None:
+    """Test that the user step uses the correct Schema."""
+    result: dict[str, Any] = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    assert result["data_schema"] == vol.Schema(
+        {
+            vol.Required(CONF_HOST): cv.string,
+            vol.Required(CONF_PASSWORD): cv.string,
+        }
+    )
+
+
+async def test_user_connection_error(hass: HomeAssistant) -> None:
+    """Test starting a flow by user but with an unexpected exception."""
+
+    with patch(
+        PYVLX_CONNECT_FUNCTION_PATH, side_effect=PyVLXException("DUMMY")
+    ) as connect_mock:
+        result: dict[str, Any] = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=deepcopy(DUMMY_DATA)
+        )
+
+        connect_mock.assert_called_once()
+
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["step_id"] == "user"
+        assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_user_unknown_error(hass: HomeAssistant) -> None:
+    """Test starting a flow by user but with an unexpected exception."""
+    with patch(
+        PYVLX_CONNECT_FUNCTION_PATH, side_effect=Exception("DUMMY")
+    ) as connect_mock:
+        result: dict[str, Any] = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=deepcopy(DUMMY_DATA)
+        )
+
+        connect_mock.assert_called_once()
+
+        assert result["type"] == data_entry_flow.FlowResultType.ABORT
+        assert result["reason"] == "unknown"
+
+
+async def test_user_success(hass: HomeAssistant) -> None:
+    """Test starting a flow by user with valid values."""
+    with patch(PYVLX_CONNECT_FUNCTION_PATH) as connect_mock, patch(
+        PYVLX_DISCONNECT_FUNCTION_PATH
+    ) as disconnect_mock:
```

Comment:
The tests are failing due to an issue in PyVLX: the heartbeat is created during `__init__` but I think it should probably be created after a successful connect instead.
Anyway, instead of creating a real instance of PyVLX and patching the two methods, maybe it would be better to patch the whole class with auto-spec.

Reply:
I tried to mock the whole class, but for some reason my attempt doesn't work. So I directly mocked the __init__ function, also without success.

Reply:
@epenet 
I mocked the whole class locally with auto-spec, but then the tests already failed local. Just mocking the __init__ doesn't solve the problem either, just like mocking each function of PyVLX separately. 

Do you have a tip how else to solve the problem.

Reply:
If you have access to `PyVLX`, then you could fix it there and move the heartbeat initialisation out of `__init__`.
Then you do not need to worry so much about mocking this part or that part in HA.